frontend/packages/dev-console/src/utils/pipeline-plugin-utils.ts

```diff
@@ -0,0 +1,71 @@
+import * as _ from 'lodash';
+import { referenceForModel, K8sResourceKind } from '@console/internal/module/k8s';
+import { FirehoseResource } from '@console/internal/components/utils';
+import { PipelineRunModel, PipelineModel } from '../models';
+import { Pipeline, PipelineRun } from './pipeline-augment';
+
+// label to get the pipelines
+export const PIPELINE_REF = 'app.openshift.io/pipeline-ref';
+
+export const tknPipelineAndPipelineRunsResources = (namespace: string): FirehoseResource[] => {
+  const resources = [
+    {
+      isList: true,
+      kind: referenceForModel(PipelineRunModel),
+      namespace,
+      prop: 'pipelineRuns',
+      optional: true,
+    },
+    {
+      isList: true,
+      kind: referenceForModel(PipelineModel),
+      namespace,
+      prop: 'pipelines',
+      optional: true,
+    },
+  ];
+  return resources;
+};
+
+type PipelineItem = {
+  pipeline: K8sResourceKind;
+  pipelineRuns: K8sResourceKind[];
+};
+
+const byCreationTime = (left, right) => {
+  const leftCreationTime = new Date(_.get(left, ['metadata', 'creationTimestamp'], Date.now()));
+  const rightCreationTime = new Date(_.get(right, ['metadata', 'creationTimestamp'], Date.now()));
+  return rightCreationTime.getTime() - leftCreationTime.getTime();
+};
+
+const getPipelineRunsForPipeline = (pipeline: Pipeline, props): PipelineRun[] => {
+  if (!props || !props.pipelineRuns) return null;
+  const pipelineRunsData = props.pipelineRuns.data;
+  const PIPELINE_RUN_LABEL = 'tekton.dev/pipeline';
+  const pipelineName = pipeline.metadata.name;
+  return pipelineRunsData
+    .filter((pr: PipelineRun) => {
+      return (
+        pipelineName ===
+        (_.get(pr, ['spec', 'pipelineRef', 'name'], null) ||
+          _.get(pr, ['metadata', 'labels', PIPELINE_RUN_LABEL], null))
+      );
+    })
+    .sort(byCreationTime);
+};
+
+export const getPipelinesAndPipelineRunsForResource = (
+  resource: K8sResourceKind,
+  props,
+): PipelineItem => {
+  if (!props || !props.pipelines || !props.pipelines.data.length) return null;
+  const pipelinesData = props.pipelines.data;
+  const pipelineName = _.get(resource, ['metadata', 'labels', PIPELINE_REF], null);
+  if (!pipelineName) return null;
+  const resourcePipeline = pipelinesData.find((pl) => pl.metadata.name === pipelineName);
+  const pipelinesAndPipelineRuns = {
+    pipeline: resourcePipeline,
+    pipelineRuns: getPipelineRunsForPipeline(resourcePipeline, props),
+  };
+  return pipelinesAndPipelineRuns;
```

Comment:
```suggestion
  return { ... };
```
No need to variable then return.